pkg/github/pullrequests.go

```diff
@@ -1148,101 +1273,459 @@ func CreatePullRequest(getClient GetClientFn, t translations.TranslationHelperFu
 			),
-			mcp.WithBoolean("maintainer_can_modify",
-				mcp.Description("Allow maintainer edits"),
+			mcp.WithString("side",
+				mcp.Description("The side of the diff to comment on"),
+				mcp.Enum("LEFT", "RIGHT"),
+			),
+			mcp.WithNumber("startLine",
+				mcp.Description("For multi-line comments, the first line of the range that the comment applies to"),
+			),
+			mcp.WithString("startSide",
+				mcp.Description("For multi-line comments, the starting side of the diff that the comment applies to"),
+				mcp.Enum("LEFT", "RIGHT"),
 			),
 		),
 		func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
 			owner, err := requiredParam[string](request, "owner")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
+
 			repo, err := requiredParam[string](request, "repo")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
-			title, err := requiredParam[string](request, "title")
+
+			pullNumber, err := requiredParam[constrainableInt32](request, "pullNumber")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
-			head, err := requiredParam[string](request, "head")
+
+			path, err := requiredParam[string](request, "path")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
-			base, err := requiredParam[string](request, "base")
+
+			body, err := requiredParam[string](request, "body")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			body, err := OptionalParam[string](request, "body")
+			subjectType, err := requiredParam[string](request, "subjectType")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			draft, err := OptionalParam[bool](request, "draft")
+			line, err := OptionalParam[constrainableInt32](request, "line")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			maintainerCanModify, err := OptionalParam[bool](request, "maintainer_can_modify")
+			side, err := OptionalParam[string](request, "side")
 			if err != nil {
 				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			newPR := &github.NewPullRequest{
-				Title: github.Ptr(title),
-				Head:  github.Ptr(head),
-				Base:  github.Ptr(base),
+			startLine, err := OptionalParam[constrainableInt32](request, "startLine")
+			if err != nil {
+				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			if body != "" {
-				newPR.Body = github.Ptr(body)
+			startSide, err := OptionalParam[string](request, "startSide")
+			if err != nil {
+				return mcp.NewToolResultError(err.Error()), nil
 			}
 
-			newPR.Draft = github.Ptr(draft)
-			newPR.MaintainerCanModify = github.Ptr(maintainerCanModify)
+			client, err := getGQLClient(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get GitHub GQL client: %w", err)
+			}
```

Comment:
TODO: A bit more e2e validation (and probably a manual check) of these working as expected